internal/util/secretskms.go

```diff
@@ -91,22 +96,55 @@ var _ = RegisterKMSProvider(KMSProvider{
 	Initializer: initSecretsMetadataKMS,
 })
 
-// initSecretsMetadataKMS initializes a SecretsMetadataKMS that wraps a
-// SecretsKMS, so that the passphrase from the StorageClass secrets can be used
+// initSecretsMetadataKMS initializes a SecretsMetadataKMS that wraps a SecretsKMS,
+// so that the passphrase from the user provided or StorageClass secrets can be used
 // for encrypting/decrypting DEKs that are stored in a detached DEKStore.
 func initSecretsMetadataKMS(args KMSInitializerArgs) (EncryptionKMS, error) {
-	eKMS, err := initSecretsKMS(args.Secrets)
-	if err != nil {
+	var smKMS SecretsMetadataKMS
+	var (
+		secretName      string
+		secretNamespace string
+	)
+
+	err := setConfigString(&secretName, args.Config, metadataSecretNameKey)
+	if errors.Is(err, errConfigOptionMissing) {
+		// if userSecret option is not specified, fetch encryptionPassphrase from
+		// storageclass secrets.
+		encryptionPassphrase, ok := args.Secrets[encryptionPassphraseKey]
+		if !ok {
+			return nil, fmt.Errorf("missing %q in storageclass secret", encryptionPassphraseKey)
+		}
+		smKMS.SecretsKMS = SecretsKMS{passphrase: encryptionPassphrase}
+
+		return smKMS, nil
+	} else if err != nil {
+		return nil, err
+	}
+
+	err = setConfigString(&secretNamespace, args.Config, metadataSecretNamespaceKey)
+	// if 'secretNamespace' not specified, defaults to namespace in which PVC was created
```

Comment:
for better readbility
```suggestion
if err != nil && !errors.Is(err, errConfigOptionMissing){
 return nil,err
 }
 // if 'secretNamespace' not specified, defaults to namespace in which PVC was created
if errors.Is(err, errConfigOptionMissing) {
   secretNamespace = args.Tenant
}

```